doc/function.md

```diff
@@ -1,5 +1,255 @@
 # Function
 
-You are reading a draft of the next documentation and it's in continuous update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
+The **Function** class is a representation of the JavaScript function. A function
+is a JavaScript procedure, a set of statements that performs a task or calculates
+a value. This class provides some methods to create and exectue a function.
+
+The `Function` class inherits its behavior from the `Object` class (for more info
+see: [Object](object.md)).
+
+## Methods
+
+### Constructor
+
+Creates a new empty instance of `Function`.
+
+```cpp
+Function();
+```
+
+### Constructor
+
+Creates a new instance of `Function` object.
+
+```cpp
+Function(napi_env env, napi_value value);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the Value object.
+- `[in] value`: The C++ primitive from which to instantiate the Value. `value`
+may be any of:
+
+  - bool
+  - Any integer type
+  - Any floating point type
+  - const char* (encoded using UTF-8, null-terminated)
+  - const char16_t* (encoded using UTF-16-LE, null-terminated)
+  - std::string (encoded using UTF-8)
+  - std::u16string
+  - napi::Value
+  - napi_value
+
+Returns a non-empty `Function` instance.
+
+### New
+
+Creates instance of a `Function` object.
+
+```cpp
+template <typename Callable>
+static Function New(napi_env env, Callable cb, const char* utf8name = nullptr, void* data = nullptr);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the Function object.
+- `[in] cb`: Object that implement the operator **()** accepting a `const CallbackInfo&`
+and returns either void or Value.
+- `[in] utf8name`: Null-terminated string to be used as the name of the function.
+- `[in] data`: User-provided data context. This will be passed back into the
+function when invoked later.
+
+Returns instance of a `Function` object.
+
+### New
+
+```cpp
+template <typename Callable>
+static Function New(napi_env env, Callable cb, const std::string& utf8name, void* data = nullptr);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the Function object.
+- `[in] cb`: Object that implement the operator **()** accepting a `const CallbackInfo&`
+and returns either void or Value.
+- `[in] utf8name`: String to be used as the name of the function.
+- `[in] data`: User-provided data context. This will be passed back into the
+function when invoked later.
+
+Returns instance of a `Function` object.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(const std::initializer_list<napi_value>& args) const;
+```
+
+- `[in] args`: Initializer list of JavaScript values as `napi_value` representing
+the arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
```

Comment:
"Returns a `Napi::Value` representing the JavaScript value..." we shouldn't say "JavaScript object" because that has a special meaning.

Reply:
Done.

Reply:
NP 🙂 Weird though ... github hasn't updated to show that 😕

Reply:
Oh, I see - you fixed the ones below, but you missed this one. NP, but please say "&#96;Napi::Value&#96;" rather than merely "&#96;Value&#96;" – that is, please use the name space with the type name.